src/main/java/cinnamon/world/items/BubbleGun.java

```diff
@@ -38,6 +48,15 @@ public void attack(Entity source) {
 
         //add
         source.getWorld().addParticle(particle);
+        if (shoot_loop == null || shoot_loop.isRemoved())
+            shoot_loop = source.getWorld().playSound(SHOOT_LOOP_SOUND, SoundCategory.ENTITY, source.getPos()).volume(1f);
```

Comment:
you forgor to add the looping flag to the sound instance

however it might create a new issue of the player moving away and the sound position staying at the start pos

if someone changes the asset to have a really long sound, that issue would be more noticeable

however again, its not a good idea to keep generic loop sounds since weapons can have varying firerates